packages/fx-core/src/component/driver/aad/utility/aadAppClient.ts

```diff
@@ -53,25 +55,38 @@ export class AadAppClient {
   @hooks([ErrorContextMW({ source: "Graph", component: "AadAppClient" })])
   public async createAadApp(
     displayName: string,
-    signInAudience = SignInAudience.AzureADMyOrg
+    signInAudience = SignInAudience.AzureADMyOrg,
+    logProvider?: LogProvider
   ): Promise<AADApplication> {
     const requestBody: IAADDefinition = {
       displayName: displayName,
       signInAudience: signInAudience,
     }; // Create an AAD app without setting anything
 
+    logProvider?.debug(
+      getLocalizedString("core.common.SentApiRequest", `${this.baseUrl}/applications`, JSON.stringify(requestBody))
+    );
     const response = await this.axios.post("applications", requestBody);
+    logProvider?.debug(
```

Comment:
axios will throw exception when received 4xx/5xx response. May need to wrap the log in a try catch block

Reply:
By the way, a minor improvement: axios has interceptors (middlewares), so maybe we can add request/response interceptor to  log the requests and reduce some duplicate code. The try-catch block should no longer required in this case.

No need to improve in this PR. You can see which effort is smaller.

Reply:
Updated. Now using interceptors in aad clients and keep directly logging in app studio client to avoid big changes. One problem is that axios@0.21.2 has a bug that will skip response interceptors when request interceptors exists, so I updated to 0.21.4, while updating the version will cause another bug in per request try: "Object literal may only specify known properties, and '"axios-retry"' does not exist in type 'AxiosRequestConfig'.". I added following code to fix build error:
···
declare module 'axios' {
  export interface AxiosRequestConfig {
    'axios-retry'?: IAxiosRetryConfig;
  }
}
···
@blackchoey  Do you have any idea on the reason?

Reply:
@blackchoey Update version of retry axios can not fix this error, so I will keep the above code.